### DISCUSSION TITLE
How can I auto connect to the user's current network?

### DISCUSSION THREAD

**Discussion by cilphex** [wevm/wagmi#1553] [Q&A]
I am writing an app and I would like to be able to change network in Metamask and use the deployed implementations across `localhost`, `goerli`, and `mainnet`.

This mostly works fine. Where I am having trouble is with using `autoConnect: true` with `createClient`. When I use `autoConnect: true` and I am already on the localhost network in Metamask, my app loads fine. When I use autoconnect and load the app while on the `goerli` network in Metamask, I get this console error:

```
Uncaught (in promise) Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="text()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0)
```

Now, I think I know part of the problem. This is my chain configuration:

```
const { chains, provider } = configureChains(
  [localhost, goerli],
  [publicProvider()],
);
```

In the chain configuration above, you can see that `localhost` comes first. When I remove it, and load the app while connected to `goerli` with `autoConnect: true`, it loads up as it should.

```
// Fix for autoconnecting on goerli.
// But I still want to autoconnect on localhost, too.
const { chains, provider } = configureChains(
  [goerli],
  [publicProvider()],
);
```

So the problem seems to be that `createClient` is trying to connect to `localhost` when I view my app when connected to `goerli` in MetaMask. As soon as I manually switch the network, it is fixed, but not when auto connecting.

How can I fix this, and get createClient to try to connect to the network that I'm connected to in MetaMask?

**Comment by quankuiyu** [comment]
network does not support ENS (operation="getAvatar", network="matic", code=UNSUPPORTED_OPERATION, version=providers/5.7.2)

**Reply by quankuiyu** [reply]
I encountered this error while switching networks

**Reply by cilphex** [reply]
This is not an answer.

**Comment by quankuiyu** [comment]
const { chains, provider, webSocketProvider } = configureChains(
  [mainnet, sepolia],
  [publicProvider()],
)

What's wrong with that

**Reply by cilphex** [reply]
This is also not an answer and actually ignores the question I asked above.
